### ISSUE TITLE
Hexbin plot ignores matplotlib default colormap

### ISSUE THREAD

**Issue by DavidMStraub** [pandas-dev/pandas#31871]
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import matplotlib as mpl
mpl.rcParams['image.cmap'] = 'Greys'
df = pd.DataFrame(np.random.randn(1000, 2), columns=["a", "b"])
df["b"] = df["b"] + np.arange(1000)
df.plot.hexbin(x="a", y="b", gridsize=25)  # this is not gray...
```

#### Problem description

I don't know if this is intentional behaviour, but plotting a data frame as hexbin with matplotlib, the matplotlib default colormap (e.g. set by the style sheet, via `rcParams`, or whatever) is ignored due to this line:

https://github.com/pandas-dev/pandas/blob/master/pandas/plotting/_matplotlib/core.py#L1010

```python
cmap = self.colormap or "BuGn"
```

What I would expect as a user is that pandas uses the default colormap I have set rather than `"BuGn"`.

Simply changing this line to
```python
cmap = self.colormap
```
is enough (since `plt.cm.get_cmap(None)` will return the default color map). If you agree, I can submit a PR.

#### Expected Output

A gray hexbin plot.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.5.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.3.0-26-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : de_DE.UTF-8
LOCALE           : de_DE.UTF-8

pandas           : 1.0.0
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 19.3.1
setuptools       : 44.0.0
Cython           : 0.29.14
pytest           : 5.3.5
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.1
IPython          : 7.12.0
pandas_datareader: None
bs4              : 4.8.2
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.3
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : 5.3.5
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.13
tables           : 3.6.1
tabulate         : 0.8.6
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

**Comment by TomAugspurger** [comment]
Is BuGn the default if the user hasn't adjusted the rcParams? If so, your proposal sounds good.

**Comment by DavidMStraub** [comment]
No, the matplotlib default colormap since v2.0 is "viridis". So this will change the default look of a pandas hexplot for people who have not customized their colormap.

**Comment by TomAugspurger** [comment]
Mmm OK. I'm less sure about this change then. Is it possible to deprecate the current behavior? When would users see warnings, and how would they silence them?

**Comment by DavidMStraub** [comment]
Do I understand correctly that you are worried about users who are relying on pandas' hexplot method to use `BuGn` by default but who are not actually setting the colormap (neither in the hexplot call nor in the matplotlib settings)? I don't really understand why this would be a concern, but if it does concern you, you could in a first step

- check if `self.colormap` is `None` (i.e. the user hasn't passed an explicit colormap)
- check if `plt.cm.get_cmap().name` happens to equal `BuGn` (probably not)

If both are true, emit a warning that matplotlib's colormap was ignored and that this behaviour will change in the future. This could be silenced by the user by supplying an explicit colormap.

**Comment by TomAugspurger** [comment]
> users who are relying on pandas' hexplot method to use BuGn by default but who are not actually setting the colormap

Yes.

So then the question is: is this worth emitting a warning over for all users currently not setting the colormap, such that they would need to specify one to silence the warning?

**Comment by DavidMStraub** [comment]
> So then the question is: is this worth emitting a warning over for all users currently not setting the colormap, such that they would need to specify one to silence the warning?

In my opinion definitely no, since in most cases I assume that users who don't specify a cmap simply don't care (i.e. want to use the defaults).

**Comment by TomAugspurger** [comment]
Sorry if I wasn't clear. I think the choices are between deprecating with a warning, and not doing anything at all.

**Comment by DavidMStraub** [comment]
OK, it's up to you. In my opinion it *is* a bug if I set a different colormap in matplotlib (e.g. when using seaborn etc.) and pandas simply overwrites this. For all the other plot types (line plots, etc.), the matplotlib defaults are used. Since these defaults could change anytime outside of pandas' control, as a *user* I don't see the problem, but I understand as a dev you might have to take other considerations into account.
